contracts/diaspore/DebtEngine.sol

```diff
@@ -205,7 +210,81 @@ contract DebtEngine is ERC721Base {
             _tokens: paidToken
         });
     }
-    
+
+    function payBatch(
+        bytes32[] _ids,
+        uint256[] _amounts,
+        address _oracle,
+        bytes8 _currency,
+        bytes _oracleData
+    ) external returns (uint256[]) {
+        uint count = _ids.length;
+        require(count == _amounts.length, "The loans and the amounts do not correspond.");
+        require(count > 0, "There are not loans to pay.");
+
+        if (_oracle != address(0)) {
+            (uint256 rate, uint256 decimals) = IOracle(_oracle).getRate(_currency, _oracleData);
+            emit ReadedOracle(rate, decimals);
+        }
+
+        uint256[] memory pays = new uint256[](count);
+        for (uint256 i = 0; i < count; i++) {
+            (,uint256 paid) = _payBatch(_ids[i], _amounts[i], _oracle, _currency, rate, decimals);
+            pays[i] = paid;
+        }
+
+        //TODO: EVENT HERE?
+
+        return pays;
+    }
+
+    function _payBatch(
```

Comment:
This is only paying one single debt, maybe the name payBatch can be missleading. It's more like an internal pay